### ISSUE TITLE
Question on Initial Grad Norm Behavior in DiffuLLaMA Training

### ISSUE THREAD

**Issue by Facico** [HKUNLP/DiffuLLaMA#21]
Thank you for your excellent work on DiffuLLaMA. I’m currently experimenting with training MDMs by converting autoregressive (AR) models with causal attention into bidirectional diffusion-based models.

During my experiments, I noticed that when training without any mask annealing ,the gradient norm becomes extremely large in the initial training steps. This often leads to instability.

I’m wondering if you have observed similar behavior when training DiffuLLaMA (the non-annealing setting described in the paper). Did you encounter this kind of large initial gradient norm? If so, do you have any insights or techniques (e.g., annealing schedules, normalization strategies, etc.) to mitigate this issue?

**Comment by summmeer** [comment]
We didn't observe the extreme norm but there might be instability or under-optimization. I suggest keeping the annealing approach.
